cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -8,6 +8,20 @@ use crate::shared::Definitions as SharedDefinitions;
 
 #[allow(clippy::many_single_char_names)]
 pub(crate) fn define(shared: &mut SharedDefinitions, x86_instructions: &InstructionGroup) {
+    let ineg = shared.instructions.by_name("ineg");
+    let mut widen = TransformGroupBuilder::new(
+        "x86_widen",
+        r#"
+    Legalize instructions by narrowing.
+
+    Use x86-specific instructions if needed."#,
+    )
+    .isa("x86")
+    .chain_with(shared.transform_groups.by_name("widen").id);
+
+    widen.custom_legalize(ineg, "convert_ineg");
+    widen.build_and_add_to(&mut shared.transform_groups);
```

Comment:
I thought this would be sufficient to get the transformation to apply to i8/i16, but it doesn't seem to be working? I'm getting stuck debugging this because I can't seem to find the place where we apply the transformations. Maybe this is in the generated Rust file?

Reply:
nvm, I think I see the spot where x86_narrow/etc gets tied to the x86 legalization, maybe I can fix this